scripts/gha/gcs_uploader.py

```diff
@@ -0,0 +1,111 @@
+"""Command line tool for submitting artifacts to Google Cloud Storage.
+
+Example usage:
+  python gcs_uploader.py --testapp_dir <dir> --key_file <path>
+
+This will find desktop integration test artifacts in the testapp directory
+and upload them to a suitable location on GCS.
+
+The optional flag --extra_artifacts can be used to supply additional filenames
+to upload. e.g. "--extra_artifacts GoogleService-Info.plist,Info.plist"
+
+"""
+
+import os
+import subprocess
+
+from absl import app
+from absl import flags
+from absl import logging
+
+from integration_testing import gcs
+
+_DEFAULT_ARTIFACTS = (
+    "integration_test",
+    "integration_test.exe",
+    "google-services.json"
+)
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string(
+    "testapp_dir", None,
+    "This directory will be searched for artifacts to upload.")
+flags.DEFINE_string(
+    "key_file", None, "Path to key file authorizing use of the GCS bucket.")
+flags.DEFINE_list(
+    "extra_artifacts", "",
+    "Additional file names to upload. The following are always uploaded: "
+    + ",".join(_DEFAULT_ARTIFACTS))
+
+
+def main(argv):
+  if len(argv) > 1:
+    raise app.UsageError("Too many command-line arguments.")
+
+  key_file_path = _fix_path(FLAGS.key_file)
+  testapp_dir = _fix_path(FLAGS.testapp_dir)
+
+  artifacts_to_upload = set(_DEFAULT_ARTIFACTS).union(FLAGS.extra_artifacts)
+
+  logging.info(
+      "Searching for the following artifacts:\n%s",
+      "\n".join(artifacts_to_upload))
+
+  artifacts = []
+  for file_dir, _, file_names in os.walk(testapp_dir):
+    for file_name in file_names:
+      if file_name in artifacts_to_upload:
+        artifacts.append(os.path.join(file_dir, file_name))
+
+  if artifacts:
+    logging.info("Found the following artifacts:\n%s", "\n".join(artifacts))
+  else:
+    logging.info("No artifacts found to upload")
+    return
+
+  gcs.authorize_gcs(key_file_path)
+
+  gcs_prefix = "gs://%s/%s" % (gcs.PROJECT_ID, gcs.get_unique_gcs_id())
```

Comment:
Noticed this being used in test_lab.py too.
Maybe its better to make this a utility function too?

Reply:
Yeah I basically already have a function that does this in test_lab, I can move it to gcs.